src/data_model/eng2math.jl

```diff
@@ -767,7 +760,7 @@ function _map_eng2math_voltage_source!(data_math::Dict{String,<:Any}, data_eng::
                 "source_id" => "_virtual_branch.$(eng_obj["source_id"])",
                 "f_bus" => bus_obj["bus_i"],
                 "t_bus" => data_math["bus_lookup"][eng_obj["bus"]],
-                "f_connections" => collect(1:nconductors),
+                "f_connections" => eng_obj["connections"],
```

Comment:
This is inconsistent with how you defined the virtual bus. This will work fine if `connections=1:3`, but will break for example when `connections=["a","b","c"]`. I suggest reverting to the old code for this line.

Reply:
I think you have this backwards, the original code was collect(1:nconductors) which will always give integers and break if connections are strings, this change will fix that

Reply:
We construct the terminals of the virtual bus as `terminals=1:length(vm)+1`, so on the from-side of the virtual branch, the available terminals will always be integers.

Reply:
You could instead do `terminals=voltage_source["connections"]` for the virtual bus, but then it becomes more messy to add an additional terminal for the ground. I think the original code is therefore the easiest solution.

Reply:
Is there a reason why the virtual bus wouldn't be a copy of the original bus's configuration in terms of both terminals and grounding?

Reply:
This goes back to my initial comment on this thread. The real question then is, do you want the user to specify the voltage source as
`vs = Dict(bus=>"sourcebus", "connections"=[1,2,3], "vm"=>[1.0, 1.0, 1.0], "va"=>[0, -120, 120]...)`
or
`vs = Dict("sourcebus", "connections"=[1,2,3,4], "vm"=>[1.0, 1.0, 1.0, 0.0], "va"=>[0, -120, 120, 0]...)` + having to ensure that at least one of the terminals is grounded